File: A17Array1/A8PrintSubarrays.java
package A17Array1;

public class A8PrintSubarrays {
    public static void PrintSubarrays(int arr[]) {
        int ts = 0;
        for (int i = 0; i <= arr.length; i++) {
            int start = i;
            for (int j = i; j <= arr.length - 1; j++) {
                int end = j;
                for (int k = start; k <= end; k++) {
                    System.out.print(arr[k] + " ");
                }
                ts++;
                System.out.println(" ");
            }
            System.out.println(" ");

        }
        System.out.println("Total number of subarray : " + ts);
    }

    public static void main(String[] args) {
        int arr[] = { 2, 4, 6, 8, 10 };
        PrintSubarrays(arr);
    }
}

File: A17Array1/A8PrintSubarrays.js
let arr = [2, 4, 6, 8, 10];

function printSubArray(arr) {
  for (let i = 0; i < arr.length; i++) {
    for (let j = 0; j < arr.length; j++) {
      let newArr = [];
      for (let k = i; k <= j; k++) {
        newArr.push(arr[k]);
      }
      if (newArr.length) {
        console.log("newArr", newArr);
      }
    }
  }
}
printSubArray(arr);
